src/features/code_actions.zig

```diff
@@ -183,9 +183,16 @@ fn handleUnusedCapture(
     // look for next non-whitespace after last '|'. if its a '{' we can insert discards.
     // this means bare loop/switch captures (w/out curlies) aren't supported.
     var block_start = capture_loc.end + 1;
-    while (block_start < builder.handle.text.len and
-        std.ascii.isWhitespace(builder.handle.text[block_start])) : (block_start += 1)
-    {}
+    var is_comment = false;
+    while (block_start < builder.handle.text.len) : (block_start += 1)
+    {
+        //If the next two characters are // then its a comment
+        if(builder.handle.text[block_start] == '/' and builder.handle.text[block_start + 1] == '/') is_comment = true;
```

Comment:
you'll need to prevent reading oob here.

Reply:
A. Travis has a point

B. Somehow, in my mind, this implied that the check won't be done every iteration;
> - if there is a comment, wait until there is a newline before caring about whether or not there is a non-whitespace character

to illustrate:
```zig
    var in_a_comment = false;
    while (block_start < builder.handle.text.len) : (block_start += 1) {
        // if in_a_comment, only care if builder.handle.text[block_start] is == '\n'
        if (in_a_comment) {
            if (builder.handle.text[block_start] != '\n') continue; // loop until then
            in_a_comment = false;
            continue; // we already know current char is '\n'
        }
        // Fun fact: code_action are only provided if handle.tree.errors.len == 0
        // and a '/' after a capture is an error, ergo we're looking at a comment
        if (builder.handle.text[block_start] == '/') {
            in_a_comment = true;
            continue; // we know what current char is, go next
        }
        if (!std.ascii.isWhitespace(builder.handle.text[block_start])) break;
    }
```

Reply:
@nullptrdevs - unless I'm misunderstanding - does the check need to be done every iteration? Since there are no multiline comments in zig, my understanding of the problem here was that if there was a newline, we could reset the comment flag regardless of if we were already in one or not.

To your fun fact block - that is a clever trick! Would that work in the case of division? e.g.
```
if ( 1  == 1) |a|
//a
{
var b = 1/1;
}
```

it seems like technically we're not going to run into issues with the above, but searching on a single `/` seems riskier than just checking OOB + the next character. What do you think?

Reply:
OOB check added

Reply:
Added an additional iteration inside of the `is_comment = true` block so that we don't waste a loop on a character we've already checked